exec/vector/src/main/java/org/apache/drill/exec/record/metadata/VariantSchema.java

```diff
@@ -215,4 +216,19 @@ public VariantSchema copy() {
     }
     return copy;
   }
+
+  // Note: no hashCode() as these objects are not meant to be keys.
```

Comment:
I do understand you intension here, it's about contract about equals and hash code, which states `if you override equals you must override hash code`. Though you don't intend to use these class as keys, looking into your `equals` method, I would state it's not true equals, it compared only certain parts.

To avoid confusion, I suggest you create your own method for comparison and don't touch equals and hash code at all.

Reply:
@arina-ielchiieva, thanks for the second review. I removed `equals()` and went back to using the existing `isEquivalent()`, but with a revised implementation. Since I could not use all the handy `Objects` plumbing for `equals()`, I had to implement some custom stuff to do the same job. Reran all tests to verify things still work.